src/TaskLineRenderer.ts

```diff
@@ -275,52 +275,50 @@ function getTaskComponentClass(component: TaskLayoutComponent, task: Task) {
 }
 
 /**
- * The dataAttributes describe the content of the component, e.g. `data-task-priority="medium"`, `data-task-due="past-1d"` etc.
+ * The data attribute describes the content of the component, e.g. `data-task-priority="medium"`, `data-task-due="past-1d"` etc.
  */
-function getTaskDataAttributes(component: TaskLayoutComponent, task: Task) {
-    const dataAttributes: AttributesDictionary = {};
-
-    function addDateDataAttributes(date: moment.Moment | null, attributeName: string) {
-        if (date) {
-            const dateValue = dateToAttribute(date);
-            if (dateValue) {
-                dataAttributes[attributeName] = dateValue;
-            }
-        }
-    }
+function getComponentDataAttribute(component: TaskLayoutComponent, task: Task) {
+    const dataAttribute: AttributesDictionary = {};
+
+    const DataAttributeNames: { [c in TaskLayoutComponent]: string } = {
+        createdDate: 'taskCreated',
+        dueDate: 'taskDue',
+        startDate: 'taskStart',
+        scheduledDate: 'taskScheduled',
+        doneDate: 'taskDone',
+        priority: 'taskPriority',
+        description: '',
+        recurrenceRule: '',
+        blockLink: '',
+    };
```

Comment:
I understand from comments elsewhere that this is a step on the way to later improvements.

But in the meantime, please can there be some comments here to explain:

- How to decide when adding a new value whether to give it an empty string or a non-empty string
- And since there is no compiler error created if a value is added to the container but not to the switch statement, a note to update the switch statement as well would be helpful.

Reply:
Oh sure, that's a great idea. Done =)